src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,221 @@
+        username="user1",
+        domain="ldap.test",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script",
+    )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip()
+
+
+@pytest.mark.tier(0)
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__register__ipa(ipa: IPA, moduledatadir: str, testdatadir: str):
+    """
+    :title: Register a passkey with the IPA command
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use ipa command to register a FIDO2.
+        2. Check the output that contains the user key mapping data.
+    :expectedresults:
+        1. New key is registered with IPA command.
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    mapping = (
+        ipa.user("user1")
+        .add()
+        .passkey_add_register(
+            pin=123456,
+            device=f"{moduledatadir}/umockdev.device",
+            ioctl=f"{moduledatadir}/umockdev.ioctl",
+            script=f"{testdatadir}/umockdev.script",
+        )
+    )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip()
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__su(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
+    """
+    :title: Check authentication of user with LDAP and IPA
+    :setup:
+        1. Add a LDAP and IPA user with passkey_mapping.
+        2. Setup SSSD client with FIDO.
+        3. Create the recording files when authentications is successful using umockdev.
+    :steps:
+        1. Check authentication of the user.
+    :expectedresults:
+        1. Command returns 0.
+    :customerscenario: False
+    """
+
+    suffix = type(provider).__name__.lower()
+
+    with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    client.sssd.start()
+
+    assert client.auth.su.passkey(
+        username="user1",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__su_fail_pin(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
+    """
+    :title: Check authentication deny of user with LDAP and IPA with incorrect pin
+    :setup:
+        1. Add a LDAP and IPA user with passkey_mapping.
+        2. Setup SSSD client with FIDO.
+        3. Create the recording files when authentication is denied due incorrect pin provided while checking auth
```

Comment:
You aren't creating the recording file, you'd be playing them.

In any case, I'd just leave the following:
```
Authentication is denied due incorrect pin provided.
```